### ISSUE TITLE
VL53L0X publishes but with error every time

### ISSUE THREAD

**Issue by tyeth** [adafruit/Adafruit_Wippersnapper_Arduino#389]
**Describe the bug**
Just running latest version on the PicoW, using latest version of earls core for pico, loaded in uf2 mode from arduino ide v2.0.3 on mac.
 Added vl53l0x and scd41, was seeing random intermittent proximity errors interspersed with the scd41 metrics. Looked like with regular (every time) errors before the distance metric, which also lacks a newline before PUBLISHING. Verified by removing scd41 it is every time.
All metrics are published successfully.

***Arduino board***
PicoW

**To Reproduce**
1. Add vl53l0x and watch serial monitor
2. Observe similar to this in console:
```
ERROR: Failed to get proximity sensor reading!
Sensor 0x29
	Proximity: 238.00PUBLISHING -> I2C Device Sensor Event Message...
PUBLISHED!
PING!
```

**Expected behavior**
Console outputs no errors, something like 
```
Sensor 0x29
	Proximity: 238.00
PUBLISHING -> I2C Device Sensor Event Message...
PUBLISHED!
```

**Which components are connected to your device**
VL53L0X + SCD41

I've turned on debugging and it appears to be working fine, no error is generated from the driver. It looks like an artifact of the data[0] instead of distance metric, which is expected per the original pull request but maybe wippersnapper is also checking the distance/proximity event type first hence the message with WS_DEBUG being defined.
Happy to close this if that's the case.

**Comment by brentru** [comment]
I don't have a vl53l0x for testing, is it specific to this board or firmware? Or the sensor?

**Comment by tyeth** [comment]
Sensor type i think. Any distance sensor and I thought there were others, but I mistakenly gained that from the age old comments in someone elses PR, and now read it was just not available yet: 
https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/pull/289#discussion_r938273490

**Comment by brentru** [comment]
@tyeth  Was this resolved by you? I can order a vl53l0x for next week.

**Comment by tyeth** [comment]
No sorry, it involved resolving proximity type outputs which look like they should be setting a sensor type property called proximity but are currently just data[0], linked in that issue comment, and I got well distracted with tasmota.

Order one, or maybe better yet the 1X which is nearly identical and 4m range instead of 2m.

   Also consider nabbing yourself the VL53L5CX, it's the big brother, an 8x8 grid with 3D gesture sensing, has onboard processing but needs a small gesture firmware pushing to it though as part of driver initialisation so not ideal for small firmwares. It looks like there was one in the works that may have slipped,
   https://blog.adafruit.com/2021/09/14/vl53lcx-stemma-qt-breakout-design/
 I got the mini version from sparkfun, super cute and practical (**Warning** _No level shifting - its 3.3v_ )
   https://www.sparkfun.com/products/19013

**Comment by brentru** [comment]
Getting distracted is never a bad thing :) 

Ordered!

**Comment by brentru** [comment]
@tyeth Does this sensor still exhibit this behavior?

**Comment by tyeth** [comment]
Just reconnected it to an S3 devkit-c and it's returning data without errors in general, but the first event after adding the component showed the error message followed by successful reads.
![image](https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/assets/6692083/2baf0b14-6d88-4def-b7b1-46466a0916eb)

Doing a reboot showed no errors, another reboot showed two initial errors then good reads.
It's not every read at all, and could well be valid responses (the sensor can return failure to read). I've got it on my list to have a proper look another time. Closing for now
